app/src/main/java/com/brandon/forzenbook/view/composables/CreateUserComposables.kt

```diff
@@ -46,104 +62,239 @@ fun CreateAccountContent(
+                    onTextChange = { location = it },
+                    currentText = location,
+                    onMaxCharacterLength = { locationError = it },
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Go)
+                )
+                if (state.isLocationError) TextFieldErrorText(text = resources.getString(R.string.createAccountLocationError))
+                BlackButton(text = resources.getString(R.string.createAccountCreateButtonText)) {
+                    keyboardController?.hide()
+                    if (onCheckConnection()) {
+                        if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
+                            onSubmit(firstName, lastName, dateOfBirth, email, location)
+                        } else emptyFieldError = true
+                    } else isDialogOpen = true
+                }
+                if (emptyFieldError) TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+            }
         }
-        InputInfoTextField(
-            hint = resources.getString(R.string.createAccountLocationHint),
-            onTextChange = { location = it },
-            characterLimit = LOCATION_CHAR_LIMIT,
-            onMaxCharacterLength = { locationError = it },
+    )
+    if (isDialogOpen) {
+        AlertDialog(
+            title = resources.getString(R.string.errorNoInternetConnection),
+            body = resources.getString(R.string.errorConnectAndTryAgain),
+            onDismissRequest = {
+                isDialogOpen = false
+            },
         )
-        if (state.isLocationError) {
-            TextFieldErrorText(text = resources.getString(R.string.createAccountLocationError))
-        }
-        BlackButton(
-            text = resources.getString(R.string.createAccountCreateButtonText),
-        ) {
-            keyboardController?.hide()
-            if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
-                onSubmit(firstName, lastName, dateOfBirth, email, location)
-            } else {
-                emptyFieldError = true
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateAccountLoading(
+    state: Loading = Loading(),
+) {
+    var firstName by rememberSaveable { mutableStateOf(state.firstName) }
+    var lastName by rememberSaveable { mutableStateOf(state.lastName) }
+    var dateOfBirth by rememberSaveable { mutableStateOf(state.dateOfBirth) }
+    var email by rememberSaveable { mutableStateOf(state.email) }
+    var location by rememberSaveable { mutableStateOf(state.location) }
+    val navController = LocalNavController.current
+    val resources = LocalContext.current.resources
+    val calendar = Calendar.getInstance()
+    val dialog = DatePickerDialog(
```

Comment:
Don't recreate this each composition

Reply:
created a function to create this when its needed only